main.go

```diff
@@ -87,11 +83,14 @@ func main1() int {
 		// Set up the test binary as the entrypoint.
 		fmt.Sprintf("--volume=%s:/init", binary),
 		"--entrypoint=/init",
+	}
 
-		// Set up the package directory as the workdir.
-		fmt.Sprintf("--volume=%s:/pwd", wd),
-		"--workdir=/pwd",
+	// Add docker flags based on our context (module-aware, GOPATH or ad hoc mode)
+	contextDockerFlags, err := buildDockerFlags()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

Comment:
return 1?

it's probably time to split this func into a separate one that returns an error, to make error handling simpler, and avoid human error like this.

Reply:
Good catch. Will split out `mainerr` in a separate PR and then rebase this PR.

Reply:
this should be a return now